From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: deepak1556 <hop2deep@gmail.com>
Date: Sun, 1 Mar 2020 00:25:19 -0800
Subject: Remove deprecated wasm module type check

The method was deprecated in favor of {IsWasmModuleObject}.
https://chromium-review.googlesource.com/c/v8/v8/+/2033170

This is already present in Node.js v14 and can be removed when we upgrade.

diff --git a/src/node_types.cc b/src/node_types.cc
index a53bcba555448fd99d582da7dc0c7af528627ca5..e816e0af54d1cf52f2af9ded49e25625b782d5be 100644
--- a/src/node_types.cc
+++ b/src/node_types.cc
@@ -35,7 +35,7 @@ namespace {
   V(DataView)                                                                 \
   V(SharedArrayBuffer)                                                        \
   V(Proxy)                                                                    \
-  V(WebAssemblyCompiledModule)                                                \
+  V(WasmModuleObject)                                                         \
   V(ModuleNamespaceObject)                                                    \
 
 
diff --git a/test/parallel/test-util-types.js b/test/parallel/test-util-types.js
index 6a9bad016993339dccf8b90e1c0cf4e9d09adcc2..ddf1af8c9183dc61bac9a8bd662edec6459d253e 100644
--- a/test/parallel/test-util-types.js
+++ b/test/parallel/test-util-types.js
@@ -50,7 +50,7 @@ for (const [ value, _method ] of [
   [ new DataView(new ArrayBuffer()) ],
   [ new SharedArrayBuffer() ],
   [ new Proxy({}, {}), 'isProxy' ],
-  [ new WebAssembly.Module(wasmBuffer), 'isWebAssemblyCompiledModule' ],
+  [ new WebAssembly.Module(wasmBuffer), 'isWasmModuleObject' ],
 ]) {
   const method = _method || `is${value.constructor.name}`;
   assert(method in types, `Missing ${method} for ${inspect(value)}`);
